cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/output/MultipleOutputsMainOutputWrapper.java

```diff
@@ -107,24 +115,27 @@ public synchronized OutputCommitter getOutputCommitter(TaskAttemptContext contex
       Map<String, OutputCommitter> delegates = new LinkedHashMap<>();
 
       List<String> namedOutputsList = MultipleOutputs.getNamedOutputsList(context);
+      TaskAttemptContext namedContext = context;
       // if there is only 1 output configured, it is the same as the root OutputFormat, so no need to have it also in
       // the delegates; otherwise, its methods would get called more than expected.
       // If more than 1 outputs are configured, then the root OutputCommitter is a NullOutputCommitter (no-op).
       // See MapReduceRuntimeService#setOutputsIfNeeded.
-      if (namedOutputsList.size() > 1) {
-        for (String name : namedOutputsList) {
+      for (String name : namedOutputsList) {
+        // CDAP-14524 Get context without prefixes for single and multiple outputs.
+        namedContext = MultipleOutputs.getNamedTaskContext(context, name);
+        if (namedOutputsList.size() > 1) {
           Class<? extends OutputFormat> namedOutputFormatClass =
             MultipleOutputs.getNamedOutputFormatClass(context, name);
 
-          TaskAttemptContext namedContext = MultipleOutputs.getNamedTaskContext(context, name);
           OutputFormat<K, V> outputFormat =
             ReflectionUtils.newInstance(namedOutputFormatClass, namedContext.getConfiguration());
           delegates.put(name, outputFormat.getOutputCommitter(namedContext));
         }
       }
+
       // return a MultipleOutputsCommitter that commits for the root output format as well as all delegate outputformats
-      committer = new MainOutputCommitter(getRootOutputFormat(context).getOutputCommitter(context), delegates,
-                                          context);
+      committer = new MainOutputCommitter(getRootOutputFormat(namedContext).getOutputCommitter(namedContext),
```

Comment:
I'm confused, this means the root output format is the last one in the namedOutputsList? It's very brittle to rely on implementation details like this instead of a direct contract